we dont have libdl so dont try linking against it

Index: src/CMakeLists.txt
--- src/CMakeLists.txt.orig
+++ src/CMakeLists.txt
@@ -43,10 +43,6 @@ if(LibXml2_FOUND)
 	list(APPEND PUBLIC_LINK_LIBS ${LibXml2_TARGET})
 endif()
 
-if(HAVE_DLOPEN)
-	list(APPEND LINK_LIBS dl)
-endif()
-
 if(ENABLE_ADVANCED_IM)
 	list(APPEND LINK_LIBS ${XercesC_TARGET})
 	if (APPLE AND NOT IOS)
